freqtrade/exchange/bitget.py

```diff
@@ -0,0 +1,249 @@
+            PriceType.INDEX: "index_price",
+        },
+    }
+
+    _supported_trading_mode_margin_pairs: list[tuple[TradingMode, MarginMode]] = [
+        (TradingMode.FUTURES, MarginMode.ISOLATED)
+    ]
+
+    @property
+    def _ccxt_config(self) -> dict:
+        config = {}
+        if self.trading_mode == TradingMode.SPOT:
+            config.update({"options": {"defaultType": "spot"}})
+        elif self.trading_mode == TradingMode.FUTURES:
+            config.update({"options": {"defaultType": "swap"}})
+        config.update(super()._ccxt_config)
+        return config
+
+    @retrier
+    def additional_exchange_init(self) -> None:
+        try:
+            if not self._config["dry_run"]:
+                if self.trading_mode == TradingMode.FUTURES:
+                    # Set position mode to one-way (hedged = False)
+                    # self._api.set_position_mode(False, None, {'productType': 'USDT-FUTURES'})
+                    logger.info("Bitget: Position mode set to one-way.")
+
+                    # Ensure we're using the futures API
+                    self._api.options["defaultType"] = "swap"
+                else:
+                    # Ensure we're using the spot API
+                    self._api.options["defaultType"] = "spot"
+
+                # Rest of the method...
+        except ccxt.DDoSProtection as e:
+            raise DDosProtection(e) from e
+        except (ccxt.OperationFailed, ccxt.ExchangeError) as e:
+            raise TemporaryError(
+                f"Error in additional_exchange_init due to {e.__class__.__name__}. Message: {e}"
+            ) from e
+        except ccxt.BaseError as e:
+            raise OperationalException(e) from e
+
+    def _get_params(
+        self,
+        side: BuySell,
+        ordertype: str,
+        leverage: float,
+        reduceOnly: bool,
+        time_in_force: str = "GTC",
+    ) -> dict:
+        params = super()._get_params(
+            side=side,
+            ordertype=ordertype,
+            leverage=leverage,
+            reduceOnly=reduceOnly,
+            time_in_force=time_in_force,
+        )
+        if self.trading_mode == TradingMode.FUTURES and self.margin_mode:
+            params["marginMode"] = self.margin_mode.lower()
+        return params
+
+    def timeframe_to_milliseconds(self, timeframe: str) -> int:
+        return ccxt.Exchange.parse_timeframe(timeframe) * 1000
+
+    async def _async_get_historic_ohlcv(
+        self,
+        pair: str,
+        timeframe: str,
+        since_ms: int,
+        candle_type: CandleType,
+        raise_: bool = False,
+        until_ms: int | None = None,
+    ) -> OHLCVResponse:
+        try:
+            pair_data = await super()._async_get_historic_ohlcv(
+                pair, timeframe, since_ms, candle_type, raise_, until_ms
+            )
+
+            pair, _, candle_type, data, partial_candle = pair_data
+
+            if candle_type not in (CandleType.FUNDING_RATE):
```

Comment:
I'm ... not entirely sure i understand why you think `_async_get_historic_ohlcv` must be re-implemented here.

Why ain't the regular implementation enough? is bitget that special when it comes to candles?